app/scripts/app-init.js

```diff
@@ -1,13 +1,19 @@
 // This file is used only for manifest version 3
 
+// We don't usually `import` files into `app-init.js` because we need to load
+// "chunks" via `importScripts`; but in this case `promise-with-resolvers` file
+// is so small we won't ever have a problem with these two files being "split".
+// Import to set up global `Promise.withResolvers` polyfill
+import '../../shared/lib/promise-with-resolvers';
+
 // Represents if importAllScripts has been run
 // eslint-disable-next-line
 let scriptsLoadInitiated = false;
 const { chrome } = globalThis;
 const testMode = process.env.IN_TEST;
 
 /**
- * @type {globalThis.stateHooks}
+ * @type {import('../../types/global').StateHooks}
```

Comment:
turns out this wasn't working with our current version of typescript. It was returning `any` 🤷